domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/review/service/ReviewService.kt

```diff
@@ -31,4 +34,71 @@ class ReviewService(
    */
   fun getCompletedReviewsForPrisoner(prisonNumber: String): List<CompletedReview> =
     reviewPersistenceAdapter.getCompletedReviews(prisonNumber)
+
+  fun determineReviewScheduleCalculationRule(releaseDate: LocalDate?, sentenceType: SentenceType, isReAdmission: Boolean, isTransfer: Boolean): ReviewScheduleCalculationRule =
+    if (isReAdmission) {
+      ReviewScheduleCalculationRule.PRISONER_READMISSION
+    } else if (isTransfer) {
+      ReviewScheduleCalculationRule.PRISONER_TRANSFER
+    } else {
+      when (sentenceType) {
+        SentenceType.REMAND -> ReviewScheduleCalculationRule.PRISONER_ON_REMAND
+        SentenceType.CONVICTED_UNSENTENCED -> ReviewScheduleCalculationRule.PRISONER_UN_SENTENCED
+        SentenceType.INDETERMINATE_SENTENCE -> ReviewScheduleCalculationRule.INDETERMINATE_SENTENCE
+
+        else -> reviewScheduleCalculationRuleBasedOnTimeLeftToServe(releaseDate!!)
+      }
+    }
+
+  fun calculateReviewWindow(reviewScheduleCalculationRule: ReviewScheduleCalculationRule): ReviewScheduleWindow =
+    when (reviewScheduleCalculationRule) {
+      ReviewScheduleCalculationRule.PRISONER_READMISSION, ReviewScheduleCalculationRule.PRISONER_TRANSFER -> ReviewScheduleWindow.fromTodayToTenDays()
+      ReviewScheduleCalculationRule.LESS_THAN_6_MONTHS_TO_SERVE -> ReviewScheduleWindow.fromOneToThreeMonths()
+      ReviewScheduleCalculationRule.BETWEEN_6_AND_12_MONTHS_TO_SERVE, ReviewScheduleCalculationRule.PRISONER_ON_REMAND, ReviewScheduleCalculationRule.PRISONER_UN_SENTENCED -> ReviewScheduleWindow.fromTwoToThreeMonths()
+      ReviewScheduleCalculationRule.BETWEEN_12_AND_60_MONTHS_TO_SERVE -> ReviewScheduleWindow.fromFourToSixMonths()
+      ReviewScheduleCalculationRule.MORE_THAN_60_MONTHS_TO_SERVE, ReviewScheduleCalculationRule.INDETERMINATE_SENTENCE -> ReviewScheduleWindow.fromTenToTwelveMonths()
+    }
+
```

Comment:
This could be made a little more readable by doing something like: 

`private fun reviewScheduleCalculationRuleBasedOnTimeLeftToServe(releaseDate: LocalDate): ReviewScheduleCalculationRule {
    val today = LocalDate.now()
    val timeLeftToServe = Period.between(today, releaseDate)
    val monthsLeft = timeLeftToServe.toTotalMonths()
    val remainderDays = timeLeftToServe.minusMonths(monthsLeft).days

    fun isExactMonths(months: Long) = monthsLeft == months && remainderDays == 0

    return when {
        monthsLeft < 6 || isExactMonths(6) -> ReviewScheduleCalculationRule.LESS_THAN_6_MONTHS_TO_SERVE
        monthsLeft < 12 || isExactMonths(12) -> ReviewScheduleCalculationRule.BETWEEN_6_AND_12_MONTHS_TO_SERVE
        monthsLeft < 60 || isExactMonths(60) -> ReviewScheduleCalculationRule.BETWEEN_12_AND_60_MONTHS_TO_SERVE
        else -> ReviewScheduleCalculationRule.MORE_THAN_60_MONTHS_TO_SERVE
    }
}`